From c4f762e6151c246e60ce2c90fdba4bae91a356fc Mon Sep 17 00:00:00 2001
Message-Id: <c4f762e6151c246e60ce2c90fdba4bae91a356fc.1687727967.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1687727967.git.sharlatanus@gmail.com>
References: <cover.1687727967.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 21 Jun 2023 22:30:25 +0100
Subject: [PATCH 6/9] gnu: stuff: Update to 1.26.0-0.9008dc0, fix build.

* gnu/packages/astronomy.scm (stuff): Update to 1.26.0-0.9008dc0, fix build.
[source]: Swap to 'git-checkout.
[native-inputs]: Add automake, autoconf, libtool, pkg-config.
---
 gnu/packages/astronomy.scm | 40 +++++++++++++++++++++++---------------
 1 file changed, 24 insertions(+), 16 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index b5d96c932f..310a6340f5 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -1068,25 +1068,33 @@ (define-public stellarium
     (license license:gpl2+)))
 
 (define-public stuff
-  (package
-    (name "stuff")
-    (version "1.26.0")
-    (source
-     (origin
-       (method url-fetch)
-       (uri (string-append "https://www.astromatic.net/download/stuff/"
-                           "stuff-" version ".tar.gz"))
-       (sha256
-        (base32 "1syibi3b86z9pikhicvkkmgxm916j732fdiw0agw0lq6z13fdcjm"))))
-    (build-system gnu-build-system)
-    (home-page "https://www.astromatic.net/software/stuff")
-    (synopsis "Astronomical catalogue simulation")
-    (description
-     "Stuff is a program that simulates \"perfect\" astronomical catalogues.
+  ;; XXX: No version tag availalbe in GitHub.
+  ;; See: https://github.com/astromatic/stuff/issues/6
+  (let ((commit "9008dc022ef53331092da248cf0a794abd6783bf")
+        (revision "0"))
+    (package
+      (name "stuff")
+      (version (git-version "1.26.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/astromatic/stuff")
+               (commit commit)))
+         (sha256
+          (base32 "004sry5lqqm7s9x4l3agysp3n63y3ga35x1rwwda4m6dc6zvla6b"))
+         (file-name (git-file-name name version))))
+      (build-system gnu-build-system)
+      (native-inputs
+       (list automake autoconf libtool pkg-config))
+      (home-page "https://www.astromatic.net/software/stuff")
+      (synopsis "Astronomical catalogue simulation")
+      (description
+       "Stuff is a program that simulates \"perfect\" astronomical catalogues.
 It generates object lists in ASCII which can read by the SkyMaker program to
 produce realistic astronomical fields.  Stuff is part of the EFIGI development
 project.")
-    (license license:gpl3+)))
+      (license license:gpl3+))))
 
 (define-public swarp
   (package
-- 
2.40.1

